ios/SketchView/SketchView.m

```diff
@@ -52,13 +52,9 @@ -(void)setToolType:(SketchToolType) toolType
     }
 }
 
--(void)setToolColor:(NSMutableDictionary *)rgba
+-(void)setToolColor:(UIColor *)rgba
 {
-    float r = [[rgba objectForKey:@"r"] floatValue];
-    float g = [[rgba objectForKey:@"g"] floatValue];
-    float b = [[rgba objectForKey:@"b"] floatValue];
-    float a = [[rgba objectForKey:@"a"] floatValue];
-    [(PenSketchTool *)penTool setToolColor:[UIColor colorWithRed:r/255.0 green:g/255.0 blue:b/255.0 alpha:a]];
+    [(PenSketchTool *)penTool setToolColor:rgba];
```

Comment:
Two things in your comment that need attention,
> Alternatively...should it not set the same color to all tools, excluding eraser?

This we can do by having a list of available tools, and on setting a color, all the tools that implement the `ToolColor` protocol can be set.

>I don't think the color to tool mapping should be handled in the native code, that can be handled in the JS when a new tool is selected, changing the prop

This made me remember that I had thought of changing the way I'm changing properties of the tool. The sketchView actually knows only about the tools not what the properties of the tool are. Therefore I had thought of encapsulating the tool properties in the tool object. I'm open to suggestions.